tests/app/main/test_services.py

```diff
@@ -282,11 +284,121 @@ def test_should_redirect_to_login_if_not_logged_in(self, data_api_client):
                      '?next=%2Fsuppliers%2Fservices%2F123')
 
 
+class TestEditService(BaseApplicationTest):
+
+    empty_service = {
+        'services': {
+            'serviceName': 'Service name 123',
+            'status': 'published',
+            'id': '123',
+            'frameworkName': 'G-Cloud 6',
+            'supplierId': 1234,
+            'supplierName': 'We supply any',
+            'lot': 'SCS',
+        }
+    }
+
+    def setup(self):
+        super(TestEditService, self).setup()
+        with self.app.test_client():
+            self.login()
+
+    def test_questions_for_this_section_can_be_changed(self):
+        with mock.patch('app.main.views.services.data_api_client') as data_api_client:
```

Comment:
Any reason why this is not a class-level decorator instead? As far as I can see every test is using this `with` for the full method body